llvm/docs/AMDGPUUsage.rst

```diff
@@ -14074,6 +14075,2267 @@ table :ref:`amdgpu-amdhsa-memory-model-code-sequences-gfx10-gfx11-table`.
+                                                             before invalidating the
+                                                             caches. Therefore
+                                                             any following
+                                                             locations read must
+                                                             be no older than
+                                                             the value read by
+                                                             the
+                                                             fence-paired-atomic.
+
+                                                         2. ``global_inv``
+
+                                                           - Apply :ref:`amdgpu-amdhsa-memory-model-code-sequences-gfx12-scopes-table`.
+                                                           - Ensures that
+                                                             following
+                                                             loads will not see
+                                                             stale data.
+
+     **Release Atomic**
+     ------------------------------------------------------------------------------------
+     store atomic release      - singlethread - global   1. buffer/global/ds/flat_store
+                               - wavefront    - local
+                                              - generic
+     store atomic release      - workgroup    - global   1. ``global_wb scope:SCOPE_SE``
+
+                                                           - If CU wavefront execution
+                                                             mode, omit.
+                                                           - In combination with the waits
+                                                             below, ensures that all
+                                                             memory operations
+                                                             have completed at workgroup
+                                                             scope before performing the
+                                                             store that is being
+                                                             released.
+
+                                                          2. | ``s_wait_bvhcnt 0x0``
+                                                             | ``s_wait_samplecnt 0x0``
+                                                             | ``s_wait_storecnt 0x0``
+                                                             | ``s_wait_loadcnt 0x0``
+                                                             | ``s_wait_dscnt 0x0``
+                                                             | **CU wavefront execution mode:**
+                                                             | ``s_wait_dscnt 0x0``
```

Comment:
Here we do have to wait for `bvhcnt` and `samplecnt`, because we need to ensure that related loads have completed before we send out any transaction that could allow another agent to modify the underlying memory.